src/GNDStk/Component/test/write.test.cpp

```diff
@@ -26,6 +26,7 @@ SCENARIO("Component write()") {
 
          THEN ("Component's write() function gives the correct result") {
             // Component's operator<< calls its write(ostream,0)
+            njoy::GNDStk::reals << std::setprecision(8) << std::scientific;
```

Comment:
If I understand correctly, this results in every double, long double or float to be written in scientific notation using a precision of 8 digits, regardless of whether this double, long double or float is in body text or metadata?

In the prototype hierarchy you're using for this test, there is no metadata that uses a floating point type so that part does not seem to be tested through this single test.

None of the other precision setting options are tested in all of this. I trust your work, but I'd like to see proper test coverage for the different options you're giving users.